text/0000-api-evolution.md

```diff
@@ -0,0 +1,748 @@
+considered a breaking change. Crate authors should consider using Cargo
+"features" for their crate to make such use opt-in.
+
+#### Minor change: adding/removing crate dependencies.
+
+The author is not aware of any possible breakage in altering dependencies (which
+is essentially private anyway).
+
+### Modules
+
+#### Major change: renaming/moving/removing any public items.
+
+Although renaming an item might seem like a minor change, according to the
+general policy design this is not a permitted form of breakage: it's not
+possible to annotate code in advance to avoid the breakage, nor is it possible
+to prevent the breakage from affecting dependencies.
+
+Of course, much of the effect of renaming/moving/removing can be achieved by
+instead using deprecation and `pub use`, and the standard library should not be
+afraid to do so!In the long run, we should consider hiding at least some old
+deprecated items from the docs, and could even consider putting out a major
+version solely as a kind of "garbage collection" for long-deprecated APIs.
+
+#### Minor change: adding new public items.
+
+Note that adding new public items is currently a breaking change, due to glob
+imports. For example, the following snippet of code will break if the `foo`
+module introduces a public item called `bar`:
+
+```rust
+use foo::*;
+fn bar() { ... }
+```
+
+The problem here is that glob imports currently do not allow any of their
+imports to be shadowed by an explicitly-define item.
+
+There are two reasons this is considered a minor change by this RFC:
+
+1. The RFC also suggests permitting shadowing of a glob import by any explicit
+   item. This has been the intended semantics of globs, but has not been
+   implemented. The details are left to a future RFC, however.
+
+2. Even if that change were made, though, there is still the case where two glob
+   imports conflict without any explicit definition "covering" them. This is
+   permitted to produce an error under the principles of this RFC because the
+   glob imports could have been written as more explicit (expanded) `use`
+   statements. It is also plausible to do this expansion automatically for a
+   crate's dependencies, to prevent breakage in the first place.
+
+### Structs
+
+See "[Signatures in type definitions](#signatures-in-type-definitions)" for some
+general remarks about changes to the actual types in a `struct` definition.
+
+#### Major change: adding a private field when all current fields are public.
+
+This change has the effect of making external struct literals impossible to
+write, which can break code irreparably.
+
+#### Major change: adding a public field when no private field exists.
+
+This change retains the ability to use struct literals, but it breaks existing
+uses of such literals; it likewise breaks exhaustive matches against the struct.
+
+#### Minor change: adding or removing private fields when at least one already exists.
```

Comment:
> also, why is it a minor change?  Because it breaks ABI?  (It doesn't seem to qualify as a feature addition on its own; I can imagine bug-fixes that work via an added private field...)

The labeling of "minor" here is in opposition to major, but is also implicitly allowed in a patch release.